sklearn/preprocessing/_data.py

```diff
@@ -2927,7 +2927,16 @@ def _neg_log_likelihood(lmbda):
             x_trans = self._yeo_johnson_transform(x, lmbda)
             n_samples = x.shape[0]
 
-            loglike = -n_samples / 2 * np.log(x_trans.var())
+            var = x_trans.var()
+            if var < np.finfo(x_trans.dtype).eps * 10:
```

Comment:
Where is the magic number coming from? Does log complain for numbers below this?

Reply:
Pulled it out of a hat
> Does log complain for numbers below this?

I think so yeah, I'll look at it more in details tomorrow

Reply:
Yeah, that may be the wrong threshold. That is the additive threshold for `1 + eps = 1` where,
```
>>> np.finfo(np.float64).eps
2.220446049250313e-16
```
while log overflows much later than that
```
>>> np.log(1e-324)
-inf
>>> np.log(1e-323)
-743.7469247408213
```
I'm still wondering though how a variance can be zero. That should only happen if the transformed array is zero, meaning that the overflow happened somewhere before.

Though I think 2e-16 may be as good a threshold as any, this case. Not sure it makes more sense to get smaller..

It's also interesting that one tests fails because the variance was zero there with 1 sample,
```
x          = array([ 1.64644051])
x_trans    = array([ 0.42860863])
```
not sure how it was working before..

Reply:
> not sure how it was working before..

that's because there's no ZeroDivisionError, only a warning